drivers/crypto/crypto_tc_shim.c

```diff
@@ -155,6 +169,12 @@ static int do_ccm_decrypt_auth(struct cipher_ctx *ctx,
 		return -EIO;
 	}
 
+	/* Only tag_len bytes is compared in
+	 * tc_ccm_decryption_verification() before TC_CRYPTO_SUCCESS
+	 * can be returned.
+	 */
+	op->out_len = ccm_param->tag_len;
+
```

Comment:
It is not clear to me why you return only the tag length.

Reply:
Is the comment right before the assignment not clear? What should be returned here instead?

Reply:
Same counter argument as in above comment.